mobile/src/main/java/org/openhab/habdroid/model/Widget.kt

```diff
@@ -75,6 +81,43 @@ data class Widget(
         Unknown
     }
 
+    fun toChartUrl(
+        prefs: SharedPreferences,
+        random: Random,
+        width: Int,
+        height: Int = width / 2,
+        chartTheme: CharSequence?,
+        density: Int,
+        forcedPeriod: String = period,
+        forcedLegend: Boolean? = legend
+    ): String {
+        val actualDensity = density.toFloat() / prefs.getChartScalingFactor()
+        val resDivider = if (prefs.shouldRequestHighResChart()) 1 else 2
+
+        val chartUrl = Uri.Builder()
+            .path("chart")
+            .appendQueryParameter(if (item!!.type === Item.Type.Group) "groups" else "items", item!!.name)
```

Comment:
I don't really like the `!!` operator usage here. Can we fall back to a sane default (null?) if item is null?

Reply:
Currently `toChartUrl()` isn't called if `widget.item` is null, so a NPE won't occur here. If I return `null` here, I have to add two more null checks (one to each of the caller code).

Reply:
`Currently` being the keyword here. IMHO the additional null checks are better than this implicit assumption.

Reply:
Done!